tools/src/test/java/org/apache/kafka/trogdor/workload/PayloadGeneratorTest.java

```diff
@@ -123,6 +130,92 @@ public void testUniformRandomPayloadGeneratorPaddingBytes() {
         assertArrayEquals(val1End, val2End);
         assertArrayEquals(val1End, val3End);
     }
+    
+    @Test
+    public void testRandomComponentPayloadGenerator() {
+        NullPayloadGenerator nullGenerator = new NullPayloadGenerator();
+        RandomGeneratorConfig nullConfig = new RandomGeneratorConfig(50, nullGenerator);
+        
+        UniformRandomPayloadGenerator uniformGenerator =
+            new UniformRandomPayloadGenerator(5, 123, 0);
+        RandomGeneratorConfig uniformConfig = new RandomGeneratorConfig(50, uniformGenerator);
+        
+        SequentialPayloadGenerator sequentialGenerator =
+            new SequentialPayloadGenerator(4, 10);
+        RandomGeneratorConfig sequentialConfig = new RandomGeneratorConfig(75, sequentialGenerator);
+        
+        ConstantPayloadGenerator constantGenerator =
+            new ConstantPayloadGenerator(4, new byte[0]);
+        RandomGeneratorConfig constantConfig = new RandomGeneratorConfig(25, constantGenerator);
+        
+        List<RandomGeneratorConfig> components1 = new ArrayList<>(Arrays.asList(nullConfig, uniformConfig));
+        List<RandomGeneratorConfig> components2 = new ArrayList<>(Arrays.asList(sequentialConfig, constantConfig));
+        byte[] expected = new byte[4];
+
+        PayloadIterator iter = new PayloadIterator(
+            new RandomComponentPayloadGenerator(4, components1));
+        int notNull = 0;
+        int isNull = 0;
+        while (notNull < 1000 || isNull < 1000) {
```

Comment:
should this be `&&`?  As it is, this loop could terminate even if we always return null and never non-null

Reply:
I think this should be || . If we used && had isNull=1000, and notNull = 0, we would terminate, since notNull is no longer less than 1000. When I ran, it only terminated when the values were 1000 and 1001.

Reply:
oh sorry, I forgot this is the condition for staying in the loop.  my mistake.